Add APL and Protection Paladin is no longer not supported

Woliance opened the pull request:
Shield gang

Woliance committed:
Update APL

Woliance committed:
Support Prot Paldin

Woliance committed:
Update sc_paladin_protection.cpp

Woliance committed:
Quick APL update while its still a PR

Woliance committed:
Dont do PRs while you just woke up

You miss things

Woliance committed:
Seriously dont.

Woliance committed:
_

Woliance committed:
more apl things that i missed because am dumb

OrcinusDrake commented on engine/class_modules/paladin/sc_paladin_protection.cpp:
```diff
@@ -923,25 +923,36 @@ void paladin_t::generate_action_prio_list_prot()
+  def -> add_action( this, "Consecration", "if=consecration.up" );
+  def -> add_action( "Divine Toll" );
+  def -> add_action( this, "Blessed Hammer", "strikes=2.4,if=charges=3" );
+  def -> add_talent( "Ashen Hallow" );
```
Should be `add_action`, this is what's killing the checks.

OrcinusDrake commented on engine/class_modules/paladin/sc_paladin_protection.cpp:
```diff
@@ -923,25 +923,36 @@ void paladin_t::generate_action_prio_list_prot()
+  def -> add_action( this, "Avengers Shield" );
+  def -> add_action( this, "Judgment", "cycle_targets=1" );
+  def -> add_action( this, "Vanquishers Hammer" );
+  def -> add_action( this, "Consecration", "if=consecration.up" );
```
Should be a not, `if=!consecration.up`

Melekus committed:
[Paladin] Protection APL misc changes and fixes

Make sure consecration start is above consecration refresh
switch a few add_action() and add_talent() around
use the correct format for covenant abilities
simplify sotr conditions (seraphim is at the top and requires the same resources, so sotr shouldn't need to make sure that seraphim isn't ready)
remove this ugly hammer of wrath line (for now)
switch judgment from cycle_targets to target_if

Woliance committed:
Making sure Avengers Shield exists

Melekus committed:
[Paladin] Protection APL formatting and typofixing

Split the latter part of the APL away from default to have it show up below cooldowns in the generated profiles.
